src/Glpi/Search/Provider/SQLProvider.php

```diff
@@ -665,26 +665,52 @@ public static function getDefaultWhereCriteria(string $itemtype): array
                 break;
 
             case 'ProjectTask':
-                $teamtable  = 'glpi_projecttaskteams';
-                $group_criteria = [];
-                if (count($_SESSION['glpigroups'])) {
-                    $group_criteria = [
-                        "$teamtable.itemtype" => Group::class,
-                        "$teamtable.items_id" => $_SESSION['glpigroups']
+                if (!Session::haveRightsOr('project', [\Project::READALL, \Project::READMY])) {
```

Comment:
The current conditions took me a few second to understand (I was afraid some unwanted rights combinaison could go through the two if/elseif and end up without restrictions).

Maybe it would be clearer going from the highest right to the lowest ?

```php
if (read all) {
  // no condition
} elseif (read own) {
   condition
} else {
   condition = "0 === 1" // Impossible condition as a safety
}
```